src/rhsmlib/services/entitlement.py

```diff
@@ -78,14 +78,27 @@ def get_pools(self, pool_subsets=None, matches=None, pool_only=None, match_insta
             installed = products.InstalledProducts(self.cp).list(options['matches'])
             results['installed'] = [x._asdict() for x in installed]
         if 'consumed' in pool_subsets:
-            consumed = self.get_consumed_product_pools(**options)
-            results['consumed'] = [x._asdict() for x in consumed]
+            consumed = self.get_consumed_product_pools(service_level=service_level, matches=matches)
+            if pool_only:
+                results['consumed'] = [x._asdict()['pool_id'] for x in consumed]
+            else:
+                results['consumed'] = [x._asdict() for x in consumed]
         if 'available' in pool_subsets:
-            results['available'] = self.get_available_pools(**options)
+            available = self.get_available_pools(
+                show_all=show_all,
+                on_date=on_date,
+                no_overlap=no_overlap,
+                match_installed=match_installed,
+                matches=matches,
+                service_level=service_level)
+            if pool_only:
+                results['available'] = [x['id'] for x in available]
+            else:
+                results['available'] = available
 
         return results
 
-    def get_consumed_product_pools(self, service_level=None, matches=None, **kwargs):
+    def get_consumed_product_pools(self, service_level=None, matches=None, pool_only=None):
```

Comment:
It seems that the `pool_only` is not used in this method and it is not called with this argument. It is possible to remove it.